snapcraft/storeapi/assertions.py

```diff
@@ -0,0 +1,262 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import json
+import subprocess
+from copy import deepcopy
+from datetime import datetime
+
+from . import StoreClient
+from . import errors
+from . import constants
+
+
+class _BaseAssertion:
+    """Private Base class to handle assertions.
+
+    Implementations are supposed to define a class attribute to determine
+    the assertion type.
+
+    :cvar str _assertion_type: the assertion type, also treated as the endpoint
+                               for the assertion on the store and the payload
+                               header key for the returned data.
```

Comment:
I'm not sure about the rules, but I'd say that if the attribute must be manipulated by subclasses, it shouldn't be marked as internal with the _ in the front.

Reply:
I am treating it as C++, `protected`, I've seen popular projects do this.

Like 'Creating Fixtures' here https://pypi.python.org/pypi/fixtures

Since python does not enforce anything it really doesn't matter.